lib/workers/pr/changelog/source-gitlab.ts

```diff
@@ -0,0 +1,157 @@
+import URL from 'url';
+import { Release } from '../../../datasource';
+import { logger } from '../../../logger';
+import { api } from '../../../platform/gitlab/gl-got-wrapper';
+import * as globalCache from '../../../util/cache/global';
+import { regEx } from '../../../util/regex';
+import * as allVersioning from '../../../versioning';
+import { BranchUpgradeConfig } from '../../common';
+import { ChangeLogRelease, ChangeLogResult } from './common';
+import { addReleaseNotes } from './release-notes';
+
+const { get: glGot } = api;
+
+const cacheNamespace = 'changelog-gitlab-release';
+
+async function getTags(
+  endpoint: string,
+  versionScheme: string,
+  repository: string
+): Promise<string[]> {
+  logger.trace('getTags() from gitlab');
+  let url = endpoint;
+  const repoid = repository.replace(/\.git/, '').replace(/\//g, '%2F');
+  url += `projects/${repoid}/repository/tags`;
+  try {
+    const res = await glGot<{ name: string }[]>(url, {
+      paginate: true,
+    });
+
+    const tags = (res && res.body) || [];
+
+    if (!tags.length) {
+      logger.debug({ repoid }, 'repository has no Gitlab tags');
+    }
+
+    return tags.map((tag) => tag.name).filter(Boolean);
+  } catch (err) {
+    logger.info({ sourceRepo: repoid }, 'Failed to fetch Gitlab tags');
+    // istanbul ignore if
+    if (err.message && err.message.includes('Bad credentials')) {
+      logger.warn('Bad credentials triggering tag fail lookup in changelog');
+      throw err;
+    }
+    return [];
+  }
+}
+
+export async function getChangeLogJSON({
+  versioning,
+  fromVersion,
+  toVersion,
+  sourceUrl,
+  releases,
+  depName,
+  manager,
+}: BranchUpgradeConfig): Promise<ChangeLogResult | null> {
+  logger.trace('getChangeLogJSON for gitlab');
+  const version = allVersioning.get(versioning);
+  const { protocol, host, pathname } = URL.parse(sourceUrl);
+  logger.debug({ protocol, host, pathname }, 'Protocol, host, pathname');
+  const baseUrl = protocol.concat('//', host, '/');
+  const apiBaseUrl = baseUrl.concat('api/v4/');
+  const repository = pathname.slice(1).replace(/\/$/, '').replace(/\.git/, '');
```

Comment:
```suggestion
  const repository = pathname.slice(1).replace(/\/$/, '').replace(/\.git$/, '');
```

We should only replace `.git` at end, otherwise we can't support `group/repo.gittish` as repo name